.ibm/pipelines/functions.sh

```diff
@@ -44,3 +44,29 @@ cleanup_namespaces() {
         fi
     done
 }
+
+skip_if_only() {
+    echo "Checking if tests need to be executed..."
+    NAMES=$(git diff --name-only main)
```

Comment:
If the PR branch is very old compared to `main`, this would return a list of changes compared to the tip of `main`, which might return a lot of files (that might not have been really changed in the PR branch). Example with #6830 :
```
$ git diff --name-only main | wc -l
202

$ git diff --name-only main
.github/workflow-data/release-management/blog-post-header.tpl.md
Dockerfile.rhel
Makefile
build/VERSION
cmd/odo/help_test.go
docs/proposals/commands-semantic.md
docs/website/blog/2020-09-24-odo-v2.0.0-ga-release.md
docs/website/blog/2020-10-28-odo-v2.0.1-release.md
docs/website/blog/2021-05-04-odo-v2.2.0-breaking-changes.md
docs/website/blog/2021-12-14-odo-v2.4.3.md
...
```

I think it would be safer to use the merge base (best common ancestor) of the two branches:
```
$ git diff --merge-base --name-only main
tests/integration/interactive_init_test.go
```